MariaDB: switch to official image and package improvements

tuxity opened the pull request:
Hi guys,

What I've done:
- [x] Use official MariaDB docker image
- [x] MariaDB version ~~10.1.21~~ 10.1.22
- [x] Cleanup configuration (based on MySQL)
- [x] Persistant volume
- [x] External access

What is coming in another PR:
- [Galera cluster](https://github.com/tuxity/universe/compare/mariadb-10.1.21...tuxity:mariadb-galera?expand=1) 

I'm using it since few weeks now, and it's working well. I hope I didn't forget something.

tuxity commented:
ping @ryadav88

ryadav88 commented:
yes, I'll review it this week.

ryadav88 commented on repo/packages/M/mariadb/3/package.json:
```diff
@@ -0,0 +1,19 @@
+  "website": "https://mariadb.org",
+  "description": "MariaDB is one of the most popular database servers in the world. It’s made by the original developers of MySQL and guaranteed to stay open source. Notable users include Wikipedia, Facebook and Google. MariaDB is developed as open source software and as a relational database it provides an SQL interface for accessing data. The latest versions of MariaDB also include GIS and JSON ...
+  "tags": ["database", "mysql", "mariadb", "sql"],
+  "preInstallNotes": "In order for MariaDB service to start successfully it requires atleast 1 CPU and 1024MB of RAM including ports. WARNING: MariaDB on DCOS is currently in ALPHA. There may be bugs, incomplete\nfeatures, incorrect documentation, or other discrepancies.\n\nIf you didn't provide a value for `host_volume` in the CLI,\nYOUR DATA WILL NOT BE SAVED IN ANY WAY.\n",
```
Remove `WARNING: MariaDB on DCOS is currently in ALPHA. There may be bugs, incomplete\nfeatures, incorrect documentation, or other discrepancies.` and this line at the beginning of `preInstallNotes`: `This DC/OS Service is currently in preview. There may be bugs, incomplete features, incorrect documentation, or other discrepancies. Preview packages should never be used in production!`

ryadav88 commented on repo/packages/M/mariadb/3/package.json:
```diff
@@ -0,0 +1,19 @@
+{
```
Service doesnt install. Please create an example for this package here: https://github.com/dcos/examples

ryadav88 replied:
Add `minDcosReleaseVersion` property

ryadav88 replied:
any update here?

ryadav88 commented on repo/packages/M/mariadb/3/package.json:
```diff
@@ -0,0 +1,19 @@
+{
+  "packagingVersion": "3.0",
+  "name": "mariadb",
+  "version": "10.1.21",
```
PR notes and image being pulled say the version is 10.1.22

ryadav88 commented on repo/packages/M/mariadb/3/config.json:
```diff
@@ -0,0 +1,135 @@
+{
```
Please follow the same semantics as the previous mariaDB package: https://github.com/mesosphere/universe/tree/version-3.x/repo/packages/M/mariadb/2

tuxity replied:
Why I can't change it ?

tuxity commented:
@ryadav88 I've updated the PR, the installation should work now. (Env variables wasn't correctly set)

I can't follow the same semantic of the version 2, because I've cleaned it like mysql package. I will add in the next PR the support of Galera cluster.

ryadav88 commented:
@tuxity ok. Please create an example for it against https://github.com/dcos/examples repo. It is now a requirement for submitting packages for Universe.

tuxity commented:
@ryadav88 ok, I will do it this week and ping you. There is not so much to explain but okay.

ryadav88 commented:
@tuxity any update here ?

ryadav88 commented:
No update since June. Closing this PR. Please reopen if you're still working on this.
